nibabel/brainvoyager/bv.py

```diff
@@ -0,0 +1,957 @@
+        ----------
+        binaryblock : {None, string} optional
+            binary block to set into header.  By default, None, in
+            which case we insert the default empty header block
+        endianness : {None, '<','>', other endian code} string, optional
+            endianness of the binaryblock.  If None, guess endianness
+            from the data.
+        check : bool, optional
+            Whether to check content of header in initialization.
+            Default is True.
+        offset : int, optional
+            offset of the actual data into to binary file (in bytes)
+        """
+        if endianness != self.default_endianness:
+            raise BvError('BV files are always little-endian')
+        self.endianness = self.default_endianness
+        if hdr_dict is None:
+            hdr_dict = _proto2default(self.hdr_dict_proto)
+        self._hdr_dict = hdr_dict
+        if offset is None:
+            self.set_data_offset(calc_BV_header_size(
+                self.hdr_dict_proto, self._hdr_dict))
+        if 'framing_cube' in self._hdr_dict:
+            self._framing_cube = self._hdr_dict['framing_cube']
+        else:
+            self._framing_cube = self._guess_framing_cube()
+        if check:
+            self.check_fix()
+        return
+
+    @classmethod
+    def from_fileobj(klass, fileobj, endianness=default_endianness,
+                     check=True):
+        """Return read structure with given or guessed endiancode.
+
+        Parameters
+        ----------
+        fileobj : file-like object
+           Needs to implement ``read`` method
+        endianness : None or endian code, optional
+           Code specifying endianness of read data
+
+        Returns
+        -------
+        header : BvFileHeader object
+           BvFileHeader object initialized from data in fileobj
+        """
+        hdr_dict = parse_BV_header(klass.hdr_dict_proto, fileobj)
+        offset = fileobj.tell()
+        return klass(hdr_dict, endianness, check, offset)
+
+    @classmethod
+    def from_header(klass, header=None, check=False):
+        """Class method to create header from another header.
+
+        Parameters
+        ----------
+        header : ``Header`` instance or mapping
+           a header of this class, or another class of header for
+           conversion to this type
+        check : {True, False}
+           whether to check header for integrity
+
+        Returns
+        -------
+        hdr : header instance
+           fresh header instance of our own class
+        """
+        # own type, return copy
+        if type(header) == klass:
+            obj = header.copy()
+            if check:
+                obj.check_fix()
+            return obj
+        # not own type, make fresh header instance
+        obj = klass(check=check)
+        if header is None:
+            return obj
+        try:  # check if there is a specific conversion routine
+            mapping = header.as_bv_map()
```

Comment:
That sounds good to me! As I have not worked with mixins yet: could you elaborate a bit on your suggestion? I do not fully understand your code example.